### PR TITLE
Post login destination fixed

### LINKED ISSUE #132: Post-login destination
Same issue as described here
https://github.com/FNNDSC/ChRIS_ui/issues/225

### PR DESCRIPTION
**Pull request by DivyanshiSingh**
Fixes: #132 
Used react-router-dom v5 to implement features asked in [FNNDSC/ChRIS_ui#225](https://github.com/FNNDSC/ChRIS_ui/issues/225)

### CODE REVIEW

**Review by jennydaman — CHANGES_REQUESTED** [review]

**Comment by jennydaman** [comment]
https://github.com/FNNDSC/ChRIS_store_ui/tree/master and https://github.com/FNNDSC/ChRIS_store_ui/tree/pf4 are diverging, we'll fix these later. But right now one change is important to be merged from https://github.com/FNNDSC/ChRIS_store_ui/tree/master into https://github.com/FNNDSC/ChRIS_store_ui/tree/pf4: https://github.com/FNNDSC/ChRIS_store_ui/pull/126

File: src/components/Router/ProtectedRoute.js [hunk 1]
```diff
@@ -12,7 +12,11 @@ const ProtectedRoute = ({ component: Component, ...rest }) => (
       if (isLoggedIn) {
         return <Component {...props} />;
       }
-        return <Redirect to="/signin" />;
+        return <Redirect to={{
+          pathname: "/signin",
+          state: {from: props.location},
+          search: `?next=${props.location.pathname}`
```
[the remark this answers is not in the document]
Comment (DivyanshiSingh) [reply src/components/Router/ProtectedRoute.js:18]: Thanks a lot to point out the sharing link issue...I will change it to use only state.

File: src/components/SignIn/SignIn.js [hunk 2]
```diff
@@ -55,26 +55,29 @@ export class SignIn extends Component {
   handleSubmit(event) {
     const authURL = process.env.REACT_APP_STORE_AUTH_URL;
     const { username, password } = this.state;
-    const { store } = this.props;
+    const { store, location, history } = this.props;
 
     this.setState({ loading: true });
     const promise = StoreClient.getAuthToken(authURL, username, password)
       .then((token) => {
```
[the remark this answers is not in the document]
Comment (DivyanshiSingh) [reply src/components/SignIn/SignIn.js]: done

**Comment by DivyanshiSingh** [comment]
Are we waiting for few more changes in this PR @jennydaman ?

**Review by jennydaman — APPROVED** [review]

**Pull request merged**